### PR TITLE
Remove old README from BraintreeDataCollector

### PR DESCRIPTION
**Pull request by scannillo**
### Why
- Follow up PR to @jonathandowning's #575

### Summary of changes

- Remove unhelpful README from BraintreeDataCollector
- Remove README from Demo

### Checklist

- ~[ ] Added a changelog entry~

### CODE REVIEW

File: Demo/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ The current status is shown on the bottom toolbar. If you've created a payment m
 
 ### Compatibility
 
-This app should be compiled with a 8.x Base SDK (Xcode 6.x) and has a deployment target of iOS 7.0.
+This app should be compiled with a 12.x Base SDK (Xcode 12.x) and has a deployment target of iOS 12.0.
```
Comment (scannillo) [inline Demo/README.md:19]: Is this file worth keeping around?

Comment (sestevens) [reply Demo/README.md:19]: I'd be OK with removing it 🤷‍♀️

Comment (scannillo) [reply Demo/README.md]: Removed

**Review by sestevens — APPROVED** [review]
👏

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**